client-runtime/src/main/java/com/microsoft/rest/v2/RestProxy.java

```diff
@@ -317,7 +317,7 @@ private Exception instantiateUnexpectedException(SwaggerMethodParser methodParse
             final Constructor<? extends RestException> exceptionConstructor = exceptionType.getConstructor(String.class, HttpResponse.class, exceptionBodyType);
             final Object exceptionBody = responseContent == null || responseContent.isEmpty() ? null : serializer.deserialize(responseContent, exceptionBodyType, bodyEncoding(response.headers()));
 
-            result = exceptionConstructor.newInstance("Status code " + responseStatusCode + ", " + responseContent, response, exceptionBody);
+            result = exceptionConstructor.newInstance("Status code " + responseStatusCode + ", " + (exceptionBody == null ? null : responseContent), response, exceptionBody);
```

Comment:
This change is essentially saying that if `exceptionBody` is null, then we're going to throw an exception that says that the `responseContent` is null. However, that's not entirely true, because it's possible that the deserializer returned null when the `responseContent` is not null and not empty. What was your intention with this change?

Reply:
This is dealing with the fact that you can't emit a null from `HttpResponse.getBodyAsStringAsync()` or any of the other variants. Actually, the more I think about this code, the more I realize it needs to be refactored. I think this might be related to why I'm getting really big, wacky console output when a file download fails in testing.